tests/test_integrated_channels/test_integrated_channel/test_exporters/test_content_metadata.py

```diff
@@ -167,6 +168,80 @@ def test_content_exporter_failed_create_export(self, mock_get_catalog_diff, mock
+        past_transmission_to_update = ContentMetadataItemTransmission(
+            enterprise_customer=self.config.enterprise_customer,
+            plugin_configuration_id=self.config.id,
+            integrated_channel_code=self.config.channel_code(),
+            content_id=FAKE_COURSE_RUN['key'],
+            channel_metadata={},
+            content_last_changed=datetime.datetime.now() - datetime.timedelta(hours=1),
+            enterprise_customer_catalog_uuid=self.config.enterprise_customer.enterprise_customer_catalogs.first().uuid,
+            remote_created_at=datetime.datetime.utcnow(),
+        )
+        past_transmission_to_update.save()
+
+
+        mock_content_metadata_response = OrderedDict()
+        mock_content_metadata_response[FAKE_COURSE_RUN['key']] = FAKE_COURSE_RUN
+        mock_content_metadata_response[FAKE_COURSE_RUN2['key']] = FAKE_COURSE_RUN2
+        mock_api_client.return_value.get_content_metadata.return_value = list(mock_content_metadata_response.values())
+        mock_create_items = []
+        mock_delete_items = []
+        mock_matched_items = [{'content_key': FAKE_COURSE_RUN['key'], 'date_updated': datetime.datetime.now()}]
+        mock_api_client.return_value.get_catalog_diff.return_value = mock_create_items, mock_delete_items, \
+            mock_matched_items
+        exporter = ContentMetadataExporter('fake-user', self.config)
+        create_payload, update_payload, delete_payload = exporter.export(max_payload_count=1)
+
+        assert update_payload.get(FAKE_COURSE_RUN['key']).channel_metadata.get('contentId') == FAKE_COURSE_RUN['key']
+        assert not create_payload
+        assert not delete_payload
+
+        mock_api_client.return_value.get_catalog_diff.assert_called_with(
+            self.config.enterprise_customer.enterprise_customer_catalogs.first(),
+            [FAKE_COURSE_RUN['key']]
+        )
+
+        past_transmission_to_update.content_last_changed = datetime.datetime.now()
+        past_transmission_to_update.save()
+
+        past_transmission_to_delete = ContentMetadataItemTransmission(
+            enterprise_customer=self.config.enterprise_customer,
+            plugin_configuration_id=self.config.id,
+            integrated_channel_code=self.config.channel_code(),
+            content_id=FAKE_COURSE_RUN2['key'],
+            channel_metadata={},
+            content_last_changed=datetime.datetime.now() - datetime.timedelta(hours=1),
+            enterprise_customer_catalog_uuid=self.config.enterprise_customer.enterprise_customer_catalogs.first().uuid,
+            remote_created_at=datetime.datetime.utcnow(),
+        )
+        past_transmission_to_delete.save()
+
+        mock_delete_items = [{'content_key': FAKE_COURSE_RUN2['key']}]
+        mock_api_client.return_value.get_catalog_diff.return_value = mock_create_items, mock_delete_items, \
+            mock_matched_items
+
+        create_payload, update_payload, delete_payload = exporter.export(max_payload_count=1)
```

Comment:
at this point in the test, the mock diff endpoint is returning a matched item and a not found (delete) item, with the max payload count of 1, the old code was ultimately not sending the delete because the early truncate was seeing the single matched item as exhausting the may_payload_count available space, even though later in the code we'd determine that we didnt need to send the udpate.